src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -1016,6 +1017,20 @@ StringSyntax createFakeModulePath(string templateLinkExpression)
             return (SyntaxFactory.CreateStringLiteral(moduleFilePath), nestedUri);
         }
 
+        /// <summary>
+        /// Used to generate the module's templateSpec format from the id parameter.
+        /// </summary>
+        private SyntaxBase GetModuleFromId(string templateID, JObject resource)
+        {   
+            // eg for modules templateSpec - ts:<subscription-ID>/<resource-group-name>/<Template-spec-name>:<version>   
+            ResourceGroupLevelResourceId id;
+            var templateSpec = "";
+            if (ResourceGroupLevelResourceId.TryParse(templateID, out id) is not true) {
+               throw  new ConversionFailedException($"Unable to parse resourceId {templateID}", resource);
```

Comment:
This exception is going to be shown to users - could you make it more descriptive? The context of **why** we're trying to parse this resourceId may not be clear.

Suggestion of something like the following: `"Unable to interpret \"{templateID}\" as a valid template spec resource id under property {resource["name"]}.properties.templateLink.id"`

Reply:
Updated it. Thanks!!